etl/steps/data/garden/who/2022-07-17/who_vaccination.py

```diff
@@ -0,0 +1,168 @@
+import json
+from functools import reduce
+from typing import Any, List, cast
+
+import pandas as pd
+from owid import catalog
+from owid.catalog import Dataset, Table
+from owid.catalog.utils import underscore_table
+from owid.datautils import geo
+from structlog import get_logger
+
+from etl.helpers import Names
+from etl.paths import DATA_DIR
+
+N = Names(__file__)
+
+UNWPP = DATA_DIR / "garden/un/2022-07-11/un_wpp"
+
+log = get_logger()
+
+N = Names(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("who_vaccination.start")
+
+    # read dataset from meadow
+    ds_meadow = Dataset(DATA_DIR / "meadow/who/2022-07-17/who_vaccination")
+    tb_meadow = ds_meadow["who_vaccination"]
+
+    df = pd.DataFrame(tb_meadow)
+
+    log.info("who_vaccination.exclude_countries")
+    df = exclude_countries(df)
+
+    log.info("who_vaccination.harmonize_countries")
+    df = harmonize_countries(df)
+
+    ds_garden = Dataset.create_empty(dest_dir)
+    ds_garden.metadata = ds_meadow.metadata
+
+    tb_garden = clean_and_format_data(df)
+
+    pop_one_yr = get_population_one_year_olds()
+    pop_one_yr = pop_one_yr.rename(columns={"location": "country", "value": "population"}).drop(
+        columns=["metric", "variant", "sex", "age"]
+    )
+    # table = N.garden_dataset["who_vaccination"]
+    tb_garden = calculate_vaccinated_unvaccinated_population(tb_garden, pop_one_yr)
+
+    tb_garden.metadata = tb_meadow.metadata
+    ds_garden.metadata.update_from_yaml(N.metadata_path)
+    tb_garden.update_metadata_from_yaml(N.metadata_path, "who_vaccination")
+
+    ds_garden.add(tb_garden)
+    ds_garden.save()
+
+    log.info("who_vaccination.end")
+
+
+def load_excluded_countries() -> List[str]:
+    with open(N.excluded_countries_path, "r") as f:
+        data = json.load(f)
+        assert isinstance(data, list)
+    return data
+
+
+def exclude_countries(df: pd.DataFrame) -> pd.DataFrame:
+    excluded_countries = load_excluded_countries()
+    return cast(pd.DataFrame, df.loc[~df.country.isin(excluded_countries)])
+
+
+def harmonize_countries(df: pd.DataFrame) -> pd.DataFrame:
+    unharmonized_countries = df["country"]
+    df = geo.harmonize_countries(df=df, countries_file=str(N.country_mapping_path))
+
+    missing_countries = set(unharmonized_countries[df.country.isnull()])
+    if any(missing_countries):
+        raise RuntimeError(
+            "The following raw country names have not been harmonized. "
+            f"Please: (a) edit {N.country_mapping_path} to include these country "
+            f"names; or (b) add them to {N.excluded_countries_path}."
+            f"Raw country names: {missing_countries}"
+        )
+
+    return df
+
+
+def get_population_one_year_olds() -> Any:
+    un_wpp_data = catalog.Dataset(UNWPP)
+    pop = un_wpp_data["population"].reset_index()
+    pop_one_yr = pop[
+        (pop["age"] == "1") & (pop["variant"] == "estimates") & (pop["metric"] == "population") & (pop["sex"] == "all")
+    ]
```

Comment:
I would suggest doing `.reset_index(drop=True)` here, which will make `pop_one_yr` become a copy, instead of a view. This is not always necessary, but often it saves you from annoying warnings (saying that you are working on a slice of a dataframe), so I tend to just do it every time I select rows in a dataset, to avoid any issues.